vault/resource_consul_secret_backend_role_test.go

```diff
@@ -37,82 +48,183 @@ func TestConsulSecretBackendRole(t *testing.T) {
+		resource.TestCheckResourceAttr(resourceName, "consul_roles.1", "role-1"),
+		resource.TestCheckResourceAttr(resourceName, "consul_roles.2", "role-2"),
+		resource.TestCheckResourceAttr(resourceName, "service_identities.#", "2"),
+		resource.TestCheckTypeSetElemAttr(resourceName, "service_identities.*", "service-0:dc1"),
+		resource.TestCheckTypeSetElemAttr(resourceName, "service_identities.*", "service-1"),
+		resource.TestCheckResourceAttr(resourceName, "node_identities.#", "2"),
+		resource.TestCheckTypeSetElemAttr(resourceName, "node_identities.*", "server-0:dc1"),
+		resource.TestCheckTypeSetElemAttr(resourceName, "node_identities.*", "client-0:dc1"),
+	}
+
+	resource.Test(t, resource.TestCase{
+		Providers:    testProviders,
+		PreCheck:     func() { testutil.TestAccPreCheck(t) },
+		CheckDestroy: testAccConsulSecretBackendRoleCheckDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config:      testConsulSecretBackendRole_initialConfig(path, name, token, false, false),
+				ExpectError: regexp.MustCompile("Use either a policy document, a list of policies or roles, or a set of service or node identities, depending on your Consul version"),
+			},
+			{
+				Config:      testConsulSecretBackendRole_initialConfig(path, name, token, true, true),
+				ExpectError: regexp.MustCompile(`Conflicting configuration arguments`),
+			},
+			{
+				Config: testConsulSecretBackendRole_initialConfig(path, name, token, false, true),
+				Check:  resource.ComposeTestCheckFunc(createTestCheckFuncs...),
+			},
+			testutil.GetImportTestStep(resourceName, false, nil),
+			{
+				Config:      testConsulSecretBackendRole_updateConfig(path, name, token, false, false),
+				ExpectError: regexp.MustCompile("Use either a policy document, a list of policies or roles, or a set of service or node identities, depending on your Consul version"),
+			},
+			{
+				Config:      testConsulSecretBackendRole_updateConfig(path, name, token, true, true),
+				ExpectError: regexp.MustCompile(`Conflicting configuration arguments`),
+			},
+			{
+				Config: testConsulSecretBackendRole_updateConfig(path, name, token, false, true),
+				Check:  resource.ComposeTestCheckFunc(updateTestCheckFuncs...),
+			},
+			testutil.GetImportTestStep(resourceName, false, nil),
+		},
+	})
+
+	// This separate test is used to check the functionality when using the legacy policies
+	// field in the provider when using newer versions of Vault (versions 1.11 and above).
+	// Imported policies will always be in the new field consul_policies, so the import test ignores
+	// both fields but has a custom ImportStateCheck function for those values.
+	resource.Test(t, resource.TestCase{
+		Providers:    testProviders,
+		PreCheck:     func() { testutil.TestAccPreCheck(t) },
+		CheckDestroy: testAccConsulSecretBackendRoleCheckDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testConsulSecretBackendRole_initialConfig(path, name, token, true, false),
+				Check: resource.ComposeTestCheckFunc(
```

Comment:
It would be good to include any checks that are common to the various phases create/upgrade etc.